embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/config/behavior/SensitiveKeysBehaviorImpl.kt

```diff
@@ -0,0 +1,16 @@
+package io.embrace.android.embracesdk.internal.config.behavior
+
+import io.embrace.android.embracesdk.internal.config.local.SdkLocalConfig
+
+private const val SENSITIVE_KEY_MAX_SIZE = 128
+
+internal class SensitiveKeysBehaviorImpl(
+    private val localConfig: SdkLocalConfig
+) : SensitiveKeysBehavior {
+
+    // TODO: max list length
+    override fun isSensitiveKey(key: String): Boolean {
+        val keysList = localConfig.sensitiveKeysDenylist ?: return false
+        return keysList.any { it.take(SENSITIVE_KEY_MAX_SIZE).toRegex().matches(key) }
```

Comment:
This is potentially quite expensive as it constructs a regex for every entry in the list every time this function is called. This would be a good candidate for memoisation if we needed to support regex, but thankfully we don't according to the spec. Therefore It should be sufficient to just check whether the key equals any object in the list.

Reply:
we don't need to support regex? that's great 😅